mlflow/tensorflow/__init__.py

```diff
@@ -887,14 +897,109 @@ def _patch_implementation(
+                                steps = history.params["steps"]
+
+                            def _extract_n_steps(input_example_n_steps):
+                                if steps > 1:
+                                    return np.array(
+                                        [
+                                            v[0][:INPUT_EXAMPLE_SAMPLE_ROWS]
+                                            for v in input_example_n_steps.take(
+                                                1
+                                            ).as_numpy_iterator()
+                                        ]
+                                    )[0]
+                                return np.array(
+                                    [
+                                        v[0]
+                                        for v in input_example_n_steps.take(
+                                            INPUT_EXAMPLE_SAMPLE_ROWS
+                                        ).as_numpy_iterator()
+                                    ]
+                                )
+
+                            return _extract_n_steps(input_training_data)
+                        elif isinstance(input_training_data, dict):
+                            input_example_slice = {
+                                k: np.take(v, range(0, INPUT_EXAMPLE_SAMPLE_ROWS))
+                                for k, v in input_training_data.items()
+                            }
+                        elif isinstance(input_training_data, tensorflow.keras.utils.Sequence):
+                            input_example_slice = input_training_data[:][0][
+                                :INPUT_EXAMPLE_SAMPLE_ROWS
+                            ]
+
+                        else:
+                            warnings.warn(
+                                "Tensorflow keras autologging only "
+                                "supports input types of: numpy.ndarray, "
+                                "dict(<key> -> numpy.ndarray), tensorflow.data.Dataset, "
+                                "or tensorflow.keras.utils.Sequence"
+                            )
+
+                        return input_example_slice
+
+                    def _infer_model_signature(input_data_slice):
+                        try:
+                            original_stop_training = history.model.stop_training
+                            model_output = history.model.predict(input_data_slice)
+
+                            if (
+                                Version(tensorflow.__version__) <= Version("2.1.4")
+                                and original_stop_training
+                            ):
+                                # For these versions, `stop_training` flag on Model is set to False
+                                # This flag is used by the callback
+                                # (inside `_log_early_stop_callback_metrics`)
```

Comment:
nit: double backticks for method ref [970, 972, 975]

Reply:
done, thx.